src/app/heroes/heroes.component.ts

```diff
@@ -9,22 +10,23 @@ import { HEROES } from '../mock-heroes';
 })
 export class HeroesComponent implements OnInit {
 
-  heroes = HEROES;
+  heroes: Hero[] = [];
   selectedHero?: Hero;
 
-  hero: Hero = {
-    id: 1,
-    name: 'Windstorm'
-  };
-
-  constructor() { }
-
-  ngOnInit(): void {
+  constructor(private heroService: HeroService, private messageService: MessageService) {}
+  
+  getHeroes(): void {
+    this.heroService.getHeroes()
```

Comment:
avoid explicit subscribe. Look into https://angular.io/api/common/AsyncPipe 

It makes the template handle subscriptions (and importantly unsubscribe).

https://paulrohan.medium.com/angular-avoiding-subscribe-method-by-replacing-it-with-an-asynpipe-when-possible-a92c20793357

Reply:
Just converted that function to use asyncpipe via the use of an observable. Does that look right?

Reply:
async pipe usage looks good. I've added another comment for an improvement